src/osd/OpRequest.cc

```diff
@@ -128,16 +129,17 @@ void OpRequest::mark_flag_point(uint8_t flag, const char *s) {
 	     flag, s, old_flags, hit_flag_points);
 }
 
-void OpRequest::mark_flag_point_string(uint8_t flag, const string& s) {
+void OpRequest::mark_flag_point_string(uint8_t flag, const char* s) {
 #ifdef WITH_LTTNG
   uint8_t old_flags = hit_flag_points;
 #endif
   mark_event(s);
+  delayed_detail = s;
```

Comment:
This will need to be moved to mark_flag_point.  The only user of mark_flag_point_string, mark_sub_op_sent, builds a heap allocated string at run time containing the osds it's waiting on.  You can't store that pointer without storing a copy as well, so I suggest simply not populating delayed_detail in mark_flag_point_string.